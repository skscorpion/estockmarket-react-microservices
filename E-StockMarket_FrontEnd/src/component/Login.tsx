import React, { useState } from 'react';
import { connect } from 'react-redux';
import {
	Box,
	Grid,
	Link,
	Checkbox,
	FormControlLabel,
	TextField,
	CssBaseline,
	Button,
	Avatar,
	Container,
	makeStyles,
} from '@material-ui/core';
import Alert from '@material-ui/lab/Alert';
import LockOutlinedIcon from '@material-ui/icons/LockOutlined';
import Typography from '@material-ui/core/Typography';
import axios from 'axios';
import { signOut, signIn } from '../actions';

interface LoginFormValues {
	username?: string;
	password?: string;
}

function Copyright() {
	return (
		<Typography variant='body2' color='textSecondary' align='center'>
			{'Copyright Â© '}
			<Link color='inherit' href='http://localhost:3002/'>
				E-Stock Market
			</Link>{' '}
			{new Date().getFullYear()}
			{'.'}
		</Typography>
	);
}

const useStyles = makeStyles((theme) => ({
	paper: {
		marginTop: theme.spacing(8),
		display: 'flex',
		flexDirection: 'column',
		alignItems: 'center',
	},
	avatar: {
		margin: theme.spacing(1),
		backgroundColor: theme.palette.secondary.main,
	},
	form: {
		width: '100%',
		marginTop: theme.spacing(1),
	},
	submit: {
		margin: theme.spacing(3, 0, 2),
	},
	color: {
		color: 'red',
	},
}));

const Login = (props: any) => {
	const formValues = {
		username: '',
		password: '',
	};
	const classes = useStyles();
	const [state, setState] = useState({
		formData: formValues,
		isLoginFailed: false,
	});

	const onAuthChange = (isSignedIn: any) => {
		if (isSignedIn) {
			props.signIn();
		} else {
			props.signOut();
		}
	};

	// set isSignedIn as false whenever loading this component
	onAuthChange(false);

	const handleChange = (e: any) => {
		setState({
			...state,
			formData: { ...state.formData, [e.target.name]: e.target.value.trim() },
		});
	};

	function handleSubmit(event: any) {
		event.preventDefault();
		const username = state.formData.username.toLowerCase();
		if (username === 'admin' && state.formData.password === 'admin') {
			const headers = {
				'Content-Type': 'application/json',
			};
			props.history.push('/Home');
		// 	const promise = axios.get(
		// 		`http://localhost:5000/Auth?username=${username}`
		// 	);

		// 	promise
		// 		.then((value) => {
		// 			onAuthChange(true);
		// 			props.history.push('/Home');
		// 		})
		// 		.catch((error) => setState({ ...state, isLoginFailed: true }));
		} else {
			setState({ ...state, isLoginFailed: true });
		}
	}

	return (
		<Container component='main' maxWidth='xs'>
			<CssBaseline />
			<div className={classes.paper}>
				<Avatar className={classes.avatar}>
					<LockOutlinedIcon />
				</Avatar>
				<Typography component='h1' variant='h5'>
					Sign in
				</Typography>
				{state.isLoginFailed && (
					<Alert severity='warning'>Invalid username or password</Alert>
				)}
				<form className={classes.form} noValidate onSubmit={handleSubmit}>
					<TextField
						variant='outlined'
						margin='normal'
						required
						fullWidth
						id='username'
						label='User Name'
						name='username'
						autoComplete='false'
						autoFocus
						onChange={handleChange}
					/>
					<TextField
						variant='outlined'
						margin='normal'
						required
						fullWidth
						name='password'
						label='Password'
						type='password'
						id='password'
						autoComplete='current-password'
						onChange={handleChange}
					/>
					<FormControlLabel
						control={<Checkbox value='remember' color='primary' />}
						label='Remember me'
					/>
					<Button
						type='submit'
						fullWidth
						variant='contained'
						color='primary'
						className={classes.submit}
					>
						Sign In
					</Button>
					<Grid container>
						<Grid item xs>
							<Link href='#' variant='body2'>
								Forgot password?
							</Link>
						</Grid>
						<Grid item>
							{/* <Link href='#' variant='body2'>
								{"Don't have an account? Sign Up"}
							</Link> */}
						</Grid>
					</Grid>
				</form>
			</div>
			<Box mt={8}>
				<Copyright />
			</Box>
		</Container>
	);
};

export default connect(null, { signIn, signOut })(Login);
