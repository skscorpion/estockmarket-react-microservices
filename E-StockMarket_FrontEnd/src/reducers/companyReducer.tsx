import {
	ADD_COMPANY,
	SELECTED_COMPANY,
	FETCH_COMPANY,
	FETCH_COMPANY_ERROR,
	SET_COMPANIES,
} from '../actions/types';

const INITIAL_STATE = {
	loading: false,
	companies: [],
	selectedCompany: {},
	error: null,
};

export default (state: any = INITIAL_STATE, action: any) => {
	console.log('action',action)
	switch (action.type) {
		case ADD_COMPANY:
			return { ...state };
		case SELECTED_COMPANY:
			console.log('state',state)
			let company: any;
			if (action.code) {
				company = state.companies.find(
					(p: any) => p.code === action.code
				);
			} else {
				company = null;
			}
			return { ...state, selectedCompany: company };
		case FETCH_COMPANY:
			return {
				...state,
				companies: action.companies,
				loading: false,
			};
		case FETCH_COMPANY_ERROR:
			return {
				...state,
				error: action.error,
				loading: false,
			};
		case SET_COMPANIES:
			return { ...state, companies: action.companies };
		default:
			return state;
	}
};
