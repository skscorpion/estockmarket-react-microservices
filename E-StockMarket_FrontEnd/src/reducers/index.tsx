import { combineReducers } from 'redux';
import authReducer from './authReducer';
import companyReducer from './companyReducer';

export default combineReducers({
	auth: authReducer,
	company: companyReducer,
});
