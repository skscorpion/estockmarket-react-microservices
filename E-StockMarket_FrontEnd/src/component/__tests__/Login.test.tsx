import React from 'react';
import { render } from '@testing-library/react';
import Login from '../Login';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import reducers from '../../reducers';
import ReactDOM from 'react-dom';
import { shallow, mount } from 'enzyme';
// import action from '../../actions';
import configureStore from 'redux-mock-store';
import moxios from 'moxios';
// import { storeFactory } from '../../test/testUtils';

const mockStore = configureStore([]);
const store = createStore(reducers);

const setup = (initialState: any = {}) => {
	// const store = storeFactory(initialState);
	const wrapper = shallow(<Login store={store} />).dive();
	return wrapper;
};

describe('Login component- input fields', () => {
	let wrapper;

	beforeEach(() => {
		const initialState = {};
		wrapper = setup(initialState);
		moxios.install();
		moxios.stubRequest('http://localhost:5000/Auth?username=admin', {
			status: 200,
			response: { jwt: 'klfsjfkljsd' },
		});
	});

	afterEach(() => {
		moxios.uninstall();
	});
	it('username field should be there', () => {
		wrapper = setup();
		const component = wrapper.find('#username');
		expect(component).toHaveLength(1);
	});

	it('password field should be there', () => {
		wrapper = setup();
		const component = wrapper.find('#password');
		expect(component).toHaveLength(1);
	});

	it('signin button should be there', () => {
		wrapper = setup();
		const component = wrapper.find('[type="submit"]');
		expect(component).toHaveLength(1);
	});

	test('form submit', () => {
		wrapper = setup();
		const obj = {
			call() {
				console.log('testfkjdsl');
			},
		};
		const fakeEvent = { preventDefault: () => console.log('preventDefault') };

		const event = {
			preventDefault() {},
		};
		const onSubmitSpy = jest.fn();

		const component = shallow(
			<Login store={store} onSubmit={onSubmitSpy} />
		).dive();
		component.find('[type="submit"]').simulate('click');
		expect(component.find('form').length).toBe(1);
	});
});
