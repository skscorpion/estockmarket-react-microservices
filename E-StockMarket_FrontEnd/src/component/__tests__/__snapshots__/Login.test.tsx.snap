// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>H O M E --- Snapshot +++capturing Snapshot of Home 1`] = `
<main
  className="MuiContainer-root MuiContainer-maxWidthXs"
>
  <div
    className="makeStyles-paper-1"
  >
    <div
      className="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-2 MuiAvatar-colorDefault"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zM9 6c0-1.66 1.34-3 3-3s3 1.34 3 3v2H9V6zm9 14H6V10h12v10zm-6-3c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z"
        />
      </svg>
    </div>
    <h1
      className="MuiTypography-root MuiTypography-h5"
    >
      Sign in
    </h1>
    <form
      className="makeStyles-form-3"
      noValidate={true}
      onSubmit={[Function]}
    >
      <div
        className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
          data-shrink={false}
          htmlFor="username"
          id="username-label"
        >
          User Name
          <span
            aria-hidden={true}
            className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
          >
             
            *
          </span>
        </label>
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoComplete="false"
            autoFocus={true}
            className="MuiInputBase-input MuiOutlinedInput-input"
            disabled={false}
            id="username"
            name="username"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={true}
            type="text"
          />
          <fieldset
            aria-hidden={true}
            className="PrivateNotchedOutline-root-6 MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="PrivateNotchedOutline-legendLabelled-8"
            >
              <span>
                User Name
                 *
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
          data-shrink={false}
          htmlFor="password"
          id="password-label"
        >
          Password
          <span
            aria-hidden={true}
            className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
          >
             
            *
          </span>
        </label>
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoComplete="current-password"
            autoFocus={false}
            className="MuiInputBase-input MuiOutlinedInput-input"
            disabled={false}
            id="password"
            name="password"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={true}
            type="password"
          />
          <fieldset
            aria-hidden={true}
            className="PrivateNotchedOutline-root-6 MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="PrivateNotchedOutline-legendLabelled-8"
            >
              <span>
                Password
                 *
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <label
        className="MuiFormControlLabel-root"
      >
        <span
          aria-disabled={false}
          className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-10 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={null}
        >
          <span
            className="MuiIconButton-label"
          >
            <input
              className="PrivateSwitchBase-input-13"
              data-indeterminate={false}
              onChange={[Function]}
              type="checkbox"
              value="remember"
            />
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
              />
            </svg>
          </span>
        </span>
        <span
          className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
        >
          Remember me
        </span>
      </label>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-submit-4 MuiButton-containedPrimary MuiButton-fullWidth"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="submit"
      >
        <span
          className="MuiButton-label"
        >
          Sign In
        </span>
      </button>
      <div
        className="MuiGrid-root MuiGrid-container"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true"
        >
          <a
            className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-body2 MuiTypography-colorPrimary"
            href="#"
            onBlur={[Function]}
            onFocus={[Function]}
          >
            Forgot password?
          </a>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item"
        />
      </div>
    </form>
  </div>
  <div
    className="MuiBox-root MuiBox-root-14"
  >
    <p
      className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
    >
      Copyright © 
      <a
        className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorInherit"
        href="http://localhost:3002/"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        E-Stock Market
      </a>
       
      2020
      .
    </p>
  </div>
</main>
`;

exports[`My Connected React-Redux Component should render with given state from Redux store 1`] = `
<main
  className="MuiContainer-root MuiContainer-maxWidthXs"
>
  <div
    className="makeStyles-paper-1"
  >
    <div
      className="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-2 MuiAvatar-colorDefault"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zM9 6c0-1.66 1.34-3 3-3s3 1.34 3 3v2H9V6zm9 14H6V10h12v10zm-6-3c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z"
        />
      </svg>
    </div>
    <h1
      className="MuiTypography-root MuiTypography-h5"
    >
      Sign in
    </h1>
    <form
      className="makeStyles-form-3"
      noValidate={true}
      onSubmit={[Function]}
    >
      <div
        className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
          data-shrink={false}
          htmlFor="username"
          id="username-label"
        >
          User Name
          <span
            aria-hidden={true}
            className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
          >
             
            *
          </span>
        </label>
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoComplete="false"
            autoFocus={true}
            className="MuiInputBase-input MuiOutlinedInput-input"
            disabled={false}
            id="username"
            name="username"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={true}
            type="text"
          />
          <fieldset
            aria-hidden={true}
            className="PrivateNotchedOutline-root-6 MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="PrivateNotchedOutline-legendLabelled-8"
            >
              <span>
                User Name
                 *
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
          data-shrink={false}
          htmlFor="password"
          id="password-label"
        >
          Password
          <span
            aria-hidden={true}
            className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
          >
             
            *
          </span>
        </label>
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoComplete="current-password"
            autoFocus={false}
            className="MuiInputBase-input MuiOutlinedInput-input"
            disabled={false}
            id="password"
            name="password"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={true}
            type="password"
          />
          <fieldset
            aria-hidden={true}
            className="PrivateNotchedOutline-root-6 MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="PrivateNotchedOutline-legendLabelled-8"
            >
              <span>
                Password
                 *
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <label
        className="MuiFormControlLabel-root"
      >
        <span
          aria-disabled={false}
          className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-10 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={null}
        >
          <span
            className="MuiIconButton-label"
          >
            <input
              className="PrivateSwitchBase-input-13"
              data-indeterminate={false}
              onChange={[Function]}
              type="checkbox"
              value="remember"
            />
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
              />
            </svg>
          </span>
        </span>
        <span
          className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
        >
          Remember me
        </span>
      </label>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-submit-4 MuiButton-containedPrimary MuiButton-fullWidth"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="submit"
      >
        <span
          className="MuiButton-label"
        >
          Sign In
        </span>
      </button>
      <div
        className="MuiGrid-root MuiGrid-container"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true"
        >
          <a
            className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-body2 MuiTypography-colorPrimary"
            href="#"
            onBlur={[Function]}
            onFocus={[Function]}
          >
            Forgot password?
          </a>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item"
        />
      </div>
    </form>
  </div>
  <div
    className="MuiBox-root MuiBox-root-14"
  >
    <p
      className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
    >
      Copyright © 
      <a
        className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorInherit"
        href="http://localhost:3002/"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        E-Stock Market
      </a>
       
      2020
      .
    </p>
  </div>
</main>
`;
